### PR TITLE
style: use black for consistent and automated formatting

### PR DESCRIPTION
**Pull request by cjp256**
- Add test case to check formatting.

- Add a couple of ignores to flake8 that conflict with black, including
  E203 and W503. For further reading, the rationale is provided @
  https://black.readthedocs.io/en/stable/the_black_code_style.html

They also recommend to disable E501 (line-too-long), but for this PR
I simply bumped the minimum required line length after using black. We
need to revisit our approach if we want to be stricter than that.

Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [flake8]
-max-line-length = 99
+max-line-length = 107
```
Comment (facundobatista) [inline .flake8:2]: Let's fix those weird lines and keep this at 99, so we don't go to a "broken master" then to fix it.

(And I need to see that this is doable with black ;) )

File: tests/test_infra.py [hunk 2]
```diff
@@ -132,3 +138,12 @@ def test_bashcompletion_all_commands():
         real_command_names.update(cmd.name for cmd in cmds)
 
     assert completed_commands == real_command_names
+
+
+def test_black():
+    """Verify all files are nicely styled."""
+    proc = subprocess.run(
+        ["black", "--check", "--diff", "."], check=False, capture_output=True, text=True
```
Comment (facundobatista) [inline tests/test_infra.py:146]: Can we use black programatically (and not as a subscript)? this is breaking my development setup (I don't run the tests from inside the VM, so "black" is not in the path)

**Review by facundobatista — CHANGES_REQUESTED** [review]
Two details, thanks!

**Review by sergiusens — APPROVED** [review]

File: tests/test_infra.py (hunk 2 above)
Comment (cjp256) [reply tests/test_infra.py:146]: I can give it a shot.

Comment (cjp256) [reply tests/test_infra.py]: Done!

File: .flake8 (hunk 1 above)
Comment (cjp256) [reply .flake8]: Addressed https://github.com/canonical/charmcraft/pull/315/commits/c43e8a7a0834a15d992455a7f36c7d09ab7b26ec

**Review by facundobatista — APPROVED** [review]
I will hate you forever, but thanks! :)

**Comment by sergiusens** [comment]
black has cookies

**Pull request merged**